edx_exams/apps/api/utils.py

```diff
@@ -0,0 +1,31 @@
+"""
+API Utility Functions
+"""
+import logging
+from datetime import timedelta
+
+from django.utils import timezone
+
+log = logging.getLogger(__name__)
+
+
+def get_exam_attempt_time_remaining(start_time, time_limit_mins, now=None):
```

Comment:
That's a good point @Zacharis278. I'm trying to understand what would happen if either/both the start time or time_limit are null. Here are my thoughts:

- if the time limit is null, then we could consider the time limit infinite and use infinity value for that. But I worry that this only works for the current case (since it will use the shorter of the time remaining vs the expiration window)
- if the start time is null, maybe the time limit would be the same as the time limit mins or maybe the time remaining would be some large value again?
- alternatively, and maybe this is the simplest solution, we could have the function return an exception if either are null.

For now I'm going to implement assuming the last one but lemme know if you think otherwise.

Reply:
Actually what I think I'll do is raise an error if either are null and keep the expiration window the same if the error is thrown.

Reply:
for most of these types of questions my default is always to just look at what we do today and see if there's any justification for changing the behavior. In this case the remaining time is zero if they have not started or an end has not yet been set. Which, logically makes sense to me.

https://github.com/openedx/edx-proctoring/blob/0ae746c253db5910cd6f91abf60e19771102130c/edx_proctoring/utils.py#L48